src/cmakeProject.ts

```diff
@@ -271,21 +271,30 @@ export class CMakeProject {
         }
         log.debug(localize('resolving.config.preset', 'Resolving the selected configure preset'));
 
-        // We want to use the original unexpanded preset file to apply the dev env in expandConfigurePreset
+        // Need to double check this preset is valid - We want to use the original unexpanded preset file to apply the dev env in expandConfigurePreset
         // we have to first check if the preset is valid in expandedPresets since we won't be expanding the whole file here, only the path up for this preset
         if (!preset.getPresetByName(preset.configurePresets(this.folderPath), configurePreset) && !preset.getPresetByName(preset.userConfigurePresets(this.folderPath), configurePreset)) {
             return undefined;
         }
 
-        // TODO: move applyDevEnv here to decouple from expandConfigurePreset
+        const workspaceFolder = lightNormalizePath(this.folderPath || '.');
+        let expandedConfigurePreset: preset.ConfigurePreset | undefined;
 
-        // modify the preset parent environment, in certain cases, to apply the Vs Dev Env on top of process.env.
-        const expandedConfigurePreset = await preset.expandConfigurePreset(this.folderPath,
-            configurePreset,
-            lightNormalizePath(this.folderPath || '.'),
-            this.sourceDir,
-            true,
-            true);
+        const presetInherits = await preset.getConfigurePresetInherits(this.folderPath, configurePreset, true, true, undefined);
```

Comment:
This can technically happen here or within the `cmakeDriver.ts` `setConfigurePreset` or `__setConfigurePreset` methods now, but we still need to eventually put the `environment` on top of the `__parentEnvironment` for the CMakeDriver, so that we ensure it executes with the preset defined env as well as the parent env. 

Or, there is the other option of changing everywhere that we grab a "configureEnvironment", for example, the cmakeDriver.ts `getConfigureEnvironment` method, we can merge `environment` on top of `__parentEnvironment`. 

This needs to happen for all preset types.

Reply:
Looking at the code, it looks like we currently do a good job of only referencing the `environment` for use with the actual driver and environment from the helper methods in the driver, `getConfigureEnvironment`, `getCMakeBuildCommandEnvironment`, etc. This likely means that if we slightly modify this to include the `__parentEnvironment`, merging preset.environment on top of it, this would cover us without introducing complications with the cache and modifying by reference.

Reply:
done